app/assets/javascripts/admin/addon/controllers/admin-badges/show.js

```diff
@@ -1,35 +1,61 @@
-import { tracked } from "@glimmer/tracking";
+import { cached, tracked } from "@glimmer/tracking";
 import Controller, { inject as controller } from "@ember/controller";
-import { action } from "@ember/object";
-import { next } from "@ember/runloop";
+import { action, getProperties } from "@ember/object";
 import { service } from "@ember/service";
-import { observes } from "@ember-decorators/object";
+import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
-import { bufferedProperty } from "discourse/mixins/buffered-content";
 import getURL from "discourse-common/lib/get-url";
 import I18n from "discourse-i18n";
-
-const IMAGE = "image";
-const ICON = "icon";
-
-// TODO: Stop using Mixin here
-export default class AdminBadgesShowController extends Controller.extend(
-  bufferedProperty("model")
-) {
+import BadgePreviewModal from "../../components/modal/badge-preview";
+
+const FORM_FIELDS = [
+  "allow_title",
+  "multiple_grant",
+  "listable",
+  "auto_revoke",
+  "enabled",
+  "show_posts",
+  "target_posts",
+  "name",
+  "description",
+  "long_description",
+  "icon",
+  "image_upload_id",
+  "query",
+  "badge_grouping_id",
+  "trigger",
+  "badge_type_id",
+];
+
+export default class AdminBadgesShowController extends Controller {
   @service router;
+  @service toasts;
   @service dialog;
+  @service modal;
+
   @controller adminBadges;
 
-  @tracked saving = false;
-  @tracked savingStatus = "";
+  @tracked model;
+  @tracked previewLoading = false;
   @tracked selectedGraphicType = null;
+  @tracked userBadges;
+  @tracked userBadgesAll;
 
-  get badgeEnabledLabel() {
-    if (this.buffered.get("enabled")) {
-      return "admin.badges.enabled";
-    } else {
-      return "admin.badges.disabled";
+  @cached
+  get formData() {
+    const data = getProperties(this.model, ...FORM_FIELDS);
```

Comment:
FormKit only works with POJO (this is actually a limitation of immer.js but I think there's value into this anyways).